src/main/resources/lib/configFile/initIdprovider.js

```diff
@@ -0,0 +1,114 @@
+ * @param {object} [params.permissions] Id provider permissions.
+ */
+function createIdProvider(params) {
+    var bean = __.newBean('com.enonic.app.oidcidprovider.lib.configFile.CreateIdProviderHandler');
+
+    bean.name = required(params, 'name');
+    bean.displayName = nullOrValue(params.displayName);
+    bean.description = nullOrValue(params.description);
+    bean.idProviderConfig = __.toScriptValue(params.idProviderConfig);
+    bean.permissions = __.toScriptValue(params.permissions);
+
+    return __.toNativeObject(bean.createIdProvider());
+};
+
+/**
+ * Returns the list of all the id providers.
+ *
+ * @returns {object[]} Array of id providers.
+ */
+function getIdProviders() {
+    var bean = __.newBean('com.enonic.app.oidcidprovider.lib.configFile.GetIdProvidersHandler');
+    return __.toNativeObject(bean.getIdProviders());
+};
+
+
+/**
+ * Check all idproviders by name
+ *
+ * @param {Array} providers
+ * @returns {Boolean} true if exists, false if not
+ */
+function exists(providers, name) {
+    for (const count in providers) {
+        const provider = providers[count];
+        if (provider && provider.key === name) {
+            log.info(`Userstore '${name}' already exists - no autoinit.`);
+            return true;
+        }
+    }
+    log.info(`Userstore '${name}' doesn't exist yet. Will try autoinit.`);
+    return false;
+}
+
+
+
+function runInitUserStores() {
+
+    const systemIdProviders = getIdProviders();
+    const configedIdProviderNames = configFile.getAllIdProviderNames();
+
+    configedIdProviderNames.forEach(idProviderName => {
+
+        if (
+            configFile.shouldAutoInit(idProviderName) &&
+            !exists(systemIdProviders, idProviderName)
+        ) {
+            const config = configFile.getConfigForIdProvider(idProviderName);
+            const displayName = config.displayName || idProviderName;
+            const description = config.description || `${configFile.CONFIG_NAMESPACE}.${idProviderName} in ${app.config["config.filename"]}`;
+
+            const result = createIdProvider({
+                name: idProviderName,
+                displayName: displayName,
+                descripton: description,
+                idProviderConfig: {
+                    descripton: description,
+                    applicationKey: app.name,
+                    config: [], // Skipping the node-level config entirely; we're going to use the .cfg anyway (although this causes invalid config fields when viewing it in the user manager)
+                },
+                permissions: [],
+            });
+
+            if (result) {
+                log.info(`Successful autoinit, created userstore: ${JSON.stringify({
+                    name: idProviderName,
+                    displayName,
+                    description
+                })}`);
+            }
+        }
+    });
+}
+
+
+exports.initUserStores = function () {
+    taskLib.executeFunction({
```

Comment:
IMO, task is an overkill here.

Reply:
Can it actually be counterproductive by forking out and create a race condition for other components or usage that might depend on it?

Reply:
I don't think a task inside a isMaster check is needed. 
Worst case, the user store is not initialized some time after startup since its a async task.

Reply:
main.js already runs in a separate thread. @espen42 by making a task you didn't make it worse, but didn't make it better either.

Reply:
Task removed.